### DISCUSSION TITLE
ibus-mozc IME preview/pre-replacement not shown in Ghostty

### DISCUSSION THREAD

**Discussion by MakotoTheKnight** [ghostty-org/ghostty#4014] [Feature Requests, Ideas] [answered]
# Description

This is similar to https://github.com/ghostty-org/ghostty/discussions/3279, with a different IME - ibus-mozc - being used.  I chose to split this out to a separate discussion rather than +1-ing the existing one since there *could* be differences between IMEs - and some precursory reading suggests that there's a difference between ibus-mozc and fcitx5 with one favoring GNOME and the other favoring KDE - but I'm OK if this could result in this issue being merged together.

When using ibus-mozc to switch between Japanese and English, I notice that the word preview does not show up like it does with other terminals or applications.

# System Info and Version
OS: Arch Linux (up to date as of 29 December 2024)
Desktop Environment + Renderer: GNOME 47.2, Wayland (wayland 1.23.1-1)
Ghostty: ghostty 1.0.0-dev+0000000
mozc: ibus-mozc 2.30.5618.102-1, Japanese language installed

What I'm validating is the feel of typing between Japanese and English, and since I don't have a native Japanese keyboard, the IME is about the best I can do.

# Comparisons between different terminals

GNOME Terminal - behavior is as expected, with characters appearing in the terminal window even as they are swapped out for the correct ones with the IME.
- Version 3.54.2 for GNOME 47, Using VTE version 0.78.2 +BIDI +GNUTLS +ICU +SYSTEMD

https://github.com/user-attachments/assets/35aafd19-2a8f-47d9-a777-65e67101d890

Ghostty - behavior is not as expected, the IME shows the substitutions but it doesn't show the characters as I'm typing them, which leads to a feeling/impression of slowness when typing.

https://github.com/user-attachments/assets/9fe8bdb0-b1fc-4fac-9261-7c1b293e1640

---

I do want to stress that I'm not Japanese and am only learning it on the side, but I do maintain a collection of art work and files that routinely appear in Japanese, so if I have to find a file with a specific Kanji it would be surprisingly slow to type it out by hand.

**Comment by mitchellh** [comment, marked as answer]
#3628
